### PR TITLE
[WinUI] Implement PointerPressed and PointerReleased

### LINKED ISSUE #13346: Add PointerPressed and PointerReleased events and commands to PointerGestureRecognizer
### Description

Add support for PointerPressed and PointerReleased events/commands alongside the existing PointerEntered, PointerMoved, and PointerExited events/commands to simplify and enable a broader set of scenarios with a single GestureRecognizer.

### Public API Changes

```csharp
public static readonly BindableProperty PointerPressedCommandProperty = BindableProperty.Create(
    nameof(PointerPressedCommand), 
    typeof(ICommand), 
    typeof(PointerGestureRecognizer), 
    null);

public static readonly BindableProperty PointerPressedCommandParameterProperty = BindableProperty.Create(
    nameof(PointerPressedCommandParameter), 
    typeof(object), 
    typeof(PointerGestureRecognizer), null);

public static readonly BindableProperty PointerReleasedCommandProperty = BindableProperty.Create(
    nameof(PointerReleasedCommand), 
    typeof(ICommand), 
    typeof(PointerGestureRecognizer), 
    null);

public static readonly BindableProperty PointerReleasedCommandParameterProperty = BindableProperty.Create(
    nameof(PointerReleasedCommandParameter), 
    typeof(object), 
    typeof(PointerGestureRecognizer), 
    null);

public event EventHandler<PointerEventArgs>? PointerPressed;

public event EventHandler<PointerEventArgs>? PointerReleased;

public ICommand PointerPressedCommand
{
    get { return (ICommand)GetValue(PointerPressedCommandProperty); }
    set { SetValue(PointerPresssedCommandProperty, value); }
}

public ICommand PointerPressedCommandParameter
{
    get { return (ICommand)GetValue(PointerPressedCommandParameterProperty); }
    set { SetValue(PointerPressedCommandParameterProperty, value); }
}

public ICommand PointerReleasedCommand
{
    get { return (ICommand)GetValue(PointerReleasedCommandProperty); }
    set { SetValue(PointerReleasedCommandProperty, value); }
}

public ICommand PointerReleasedCommandParameter
{
    get { return (ICommand)GetValue(PointerReleasedCommandParameterProperty); }
    set { SetValue(PointerReleasedCommandParameterProperty, value); }
}

```

### Intended Use-Case

Support things like drag, resize, etc. without having to coordinate with other gesture recognizers and/or hook into native platform pointer events / gesture recognisers.

### PR DESCRIPTION
**Pull request by rachelkang**
### Description of Change

Implemented PointerPressed and PointerReleased on Windows

MacCatalyst impl to come in separate PR

### Issues Fixed

Fixes #13346

### CODE REVIEW

**Comment by jknaudt21** [comment]
Thanks for adding docs 🥇

File: src/Controls/samples/Controls.Sample/Pages/Core/PointerGestureGalleryPage.xaml.cs [hunk 1]
```diff
@@ -5,20 +5,54 @@ namespace Maui.Controls.Sample.Pages
...
+
+		void PointerPressStarted(object sender, PointerEventArgs e)
+		{
+			pgrLabel.Text = "Thanks for pressing me! Now release me!";
+			pgrLabel.BackgroundColor = Colors.SkyBlue;
+		}
+
+		void PointerPressEnded(object sender, PointerEventArgs e)
+		{
+			pgrLabel.Text = "Thanks for releasing me! Press me again or leave me!";
```
Comment (samhouts) [inline src/Controls/samples/Controls.Sample/Pages/Core/PointerGestureGalleryPage.xaml.cs:54]: don't leave me 😭

File: src/Controls/src/Core/PointerGestureRecognizer.cs [hunk 2]
```diff
@@ -40,6 +40,27 @@ public sealed class PointerGestureRecognizer : GestureRecognizer
...
+		/// </summary>
+		public static readonly BindableProperty PointerPressedCommandParameterProperty = BindableProperty.Create(nameof(PointerPressedCommandParameter), typeof(object), typeof(PointerGestureRecognizer), null);
+
+		/// <summary>
+		/// A command to invoke when the pointer that has previous initiated a press is released within the view. This is a bindable property.
+		/// </summary>
+		public static readonly BindableProperty PointerReleasedCommandProperty = BindableProperty.Create(nameof(PointerReleasedCommand), typeof(ICommand), typeof(PointerGestureRecognizer), null);
+
+		/// <summary>
+		/// An object to be passed to the PointerPressedCommand. This is a bindable property.
```
Comment (samhouts) [inline src/Controls/src/Core/PointerGestureRecognizer.cs:59]:
```suggestion
		/// An object to be passed to the PointerReleasedCommand. This is a bindable property.
```

**Review by samhouts — CHANGES_REQUESTED** [review]
I do love pasta, but not COPYPASTA! 🍝

**Review by samhouts — APPROVED** [review]

**Review by PureWeen — CHANGES_REQUESTED** [review]
Thoughts on these Remarks?

https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.pointerreleased?view=winrt-22621#remarks

**Comment by rachelkang** [comment]
> Thoughts on these Remarks?
> 
> https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.pointerreleased?view=winrt-22621#remarks

I'll test more scenarios that might be affected by this (i.e. releasing pointer over a different control). Let me know if there are other edge case scenarios that come to mind!

**Review by PureWeen — APPROVED** [review]

**Pull request merged**